### PR TITLE
#104 Fix icons: crop all to viewBox (no whitespace), #90 Fix Button icon

### LINKED ISSUE #90: Fix Button icon
doesnt work very well with custom padding, width

screenshot 2019-03-07 at 17 09 40

screenshot 2019-03-07 at 17 07 21

### LINKED ISSUE #104: Fix icons: crop all to viewBox (no whitespace)

### PR DESCRIPTION
**Pull request by keip**
Changes icons from array to object, added `viewBox` and `ratio` to /config/icons.js
The `viewBox` makes the icon 'fit in box' and  if necessary the width is calculated from the `ratio`.
fix #104, fix #90

### CODE REVIEW

**Comment by vladimirvolek** [comment]
Please fix new icon sizes in other components in storybook.

for example checkbox:

image

File: src/components/Textarea/index.js [hunk 1]
```diff
@@ -146,6 +151,12 @@ const ArrowUp = styled.div`
     z-index: 10001;
 `;
 
+const StyledIcon = styled(Icon)`
```
Comment (slowbackspace) [inline src/components/Textarea/index.js]: huh? that doesn't seems to be related to cropping icons.

File: src/components/Textarea/index.js [hunk 2]
```diff
@@ -175,26 +186,35 @@ class TextArea extends PureComponent {
...
-                    value={this.props.value}
-                    readOnly={this.props.readOnly}
-                    onClick={this.props.autoSelect ? event => event.target.select() : null}
-                    placeholder={this.props.placeholder}
-                    onChange={this.props.onChange}
-                    border={this.getColor(this.props.state)}
-                />
+                <TextAreaWrapper>
+                    {this.props.state && (
+                        <StyledIcon
```
Comment (slowbackspace) [inline src/components/Textarea/index.js]: The state icon was removed for a reason. Why did you add it back?https://github.com/trezor/trezor-ui-components/pull/86

**Review by slowbackspace — APPROVED** [review]

**Review by slowbackspace — CHANGES_REQUESTED** [review]
textarea shouldnt have state icon

**Review by slowbackspace — CHANGES_REQUESTED** [review]
Doesn't completely fix https://github.com/trezor/trezor-ui-components/issues/90
There is sill an issue when button width is set to 100%, in that case a content of the button should be centered.
Screenshot 2019-03-14 at 16 54 10

**Comment by keip** [comment]
> Doesn't completely fix #90
> There is sill an issue when button width is set to 100%, in that case a content of the button should be centered.
> Screenshot 2019-03-14 at 16 54 10

The button has `display: flex;` so if you use it with styled-component you add `width: 100%;` and `justify-content: center;`

Or we can add one more prop for buttons (fullWidth) and then you can choose the justify-content from select. What do you think?

File: src/components/Textarea/index.js (hunk 2 above)
Comment (keip) [reply src/components/Textarea/index.js]: fixed

**Comment by slowbackspace** [comment]
> > Doesn't completely fix #90
> > There is sill an issue when button width is set to 100%, in that case a content of the button should be centered.
> > Screenshot 2019-03-14 at 16 54 10
> 
> The button has `display: flex;` so if you use it with styled-component you add `width: 100%;` and `justify-content: center;`
> 
> Or we can add one more prop for buttons (fullWidth) and then you can choose the justify-content from select. What do you think?

Let's just add `justify-content: center` so if someone chooses to set custom width the content will always be centered.

**Comment by keip** [comment]
> > > Doesn't completely fix #90
> > > There is sill an issue when button width is set to 100%, in that case a content of the button should be centered.
> > > Screenshot 2019-03-14 at 16 54 10
> > 
> > 
> > The button has `display: flex;` so if you use it with styled-component you add `width: 100%;` and `justify-content: center;`
> > Or we can add one more prop for buttons (fullWidth) and then you can choose the justify-content from select. What do you think?
> 
> Let's just add `justify-content: center` so if someone chooses to set custom width the content will always be centered.

fixed

**Review by slowbackspace — APPROVED** [review]
awesome!

**Pull request merged**